doc/source/trajectories/transformations.rst

```diff
@@ -0,0 +1,87 @@
+.. -*- coding: utf-8 -*-
+.. _transformations:
+
+On-the-fly transformations
+==========================
+
+An on-the-fly transformation is a function that modifies the data contained in a trajectory :class:`~MDAnalysis.coordinates.base.Timestep`. It is called for each current time step as it is loaded into memory. A transformation function must also return the current :class:`~MDAnalysis.coordinates.base.Timestep`, as transformations are often chained together.
```

Comment:
Maybe clarify that the point is so that the transformations are applied automatically/silently as filters (though maybe this word in ambiguous) to provide the desired representation of your system.

One example of `x+2` below is good for custom plugin demonstration, but the residue one is (as far as I can tell) useless? Maybe better is to show how to unwrap every frame to rectify a simulation output for visualisation?

Reply:
I took that out and added in an example [unwrapping, entering a protein in a box, and wrapping water around it](https://lilyminium.github.io/UserGuide/trajectories/transformations.html). Unfortunately normal reStructuredText files don't let me embed NGLView widgets, but I'll be able to link to related notebooks on transformations when #38 is merged.